src/hotspot/share/compiler/disassembler.cpp

```diff
@@ -755,10 +755,10 @@ address decode_env::decode_instructions(address start, address end, address orig
 // Each method will create a decode_env before decoding.
 // You can call the decode_env methods directly if you already have one.
 
-void* Disassembler::dll_load(char* buf, int offset, size_t buflen, char* ebuf, size_t ebuflen, outputStream* st) {
-  if (offset + strlen(hsdis_library_name) + strlen(os::dll_file_extension()) < buflen) {
-    strcpy(&buf[offset], hsdis_library_name);
-    strcat(&buf[offset], os::dll_file_extension());
+void* Disassembler::dll_load(char* buf, int offset, int buflen, char* ebuf, int ebuflen, outputStream* st) {
```

Comment:
Please change the argument order - so that the lengths comes directly after the buffers. Like this:
Disassembler::dll_load(char* buf, int buflen, int offset, char* ebuf, int ebuflen, outputStream* st)